### PR TITLE
Tests for DeviceNodeGateway

### PR DESCRIPTION
**Pull request by DunderRoffe**

### CODE REVIEW

File: libsoftwarecontainer/src/gateway/devicenodegateway.cpp [hunk 1]
```diff
@@ -86,18 +91,32 @@ bool DeviceNodeGateway::activateGateway()
             parseInt(dev.mode.c_str(), &mode);
 
             // mknod dev.name c dev.major dev.minor
-            if (mknod(dev.name.c_str(), S_IFCHR | mode, makedev(majorVersion, minorVersion)) != 0) {
+            pid_t pid = INVALID_PID;
+            getContainer()->executeInContainer([&] () {
+                  return mknod(dev.name.c_str(), S_IFCHR | mode,
+                               makedev(majorVersion, minorVersion));
+                  }, &pid);
```
Comment (breakreturn) [inline libsoftwarecontainer/src/gateway/devicenodegateway.cpp]: To me the indentation of the bracket here looks misaligned. But it looks consistent with the rest so perhaps this is how we want it?

File: libsoftwarecontainer/src/gateway/devicenodegateway.h [hunk 2]
```diff
@@ -60,7 +60,7 @@
  *                        "mode":  "666"
  *                    }
  *                ]
- *  }
+ *
```
Comment (tobsan) [inline libsoftwarecontainer/src/gateway/devicenodegateway.h]: Indentation is weird here, please fix.

Comment (breakreturn) [reply libsoftwarecontainer/src/gateway/devicenodegateway.h]: jsonlint.com - paste the string and use the format the page turns it into, would make all y'all consistent

File: libsoftwarecontainer/src/gateway/devicenodegateway.cpp (hunk 1 above)
Comment (DunderRoffe) [reply libsoftwarecontainer/src/gateway/devicenodegateway.cpp]: No you are right I will change this

File: libsoftwarecontainer/src/gateway/devicenodegateway.h (hunk 2 above)
Comment (DunderRoffe) [reply libsoftwarecontainer/src/gateway/devicenodegateway.h]: I fix

**Comment by tobsan** [comment]
Ok, looks good to me.

**Pull request merged**